build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/tscTask.ts

```diff
@@ -124,13 +124,21 @@ export class TscTask extends LeafTask {
 		}
 
 		const program = tsBuildInfo.program;
+		const noEmit = config.options.noEmit ?? false;
+		const hasChangedFiles = (program.changeFileSet?.length ?? 0) > 0;
+		const hasEmitErrorsOrPending =
+			(program.affectedFilesPendingEmit?.length ??
+				program.emitDiagnosticsPerFile?.length ??
+				0) > 0;
```

Comment:
This will still return false if affectedFilesPendingEmit is 0 but emitDiagnosticsPerFile has items. I don't know if that can happen.

Reply:
Should we just check that both are not > 0, then?